### ISSUE TITLE
Can not generate graphene grain boundary with [1 1 1] rotation axis

### ISSUE THREAD

**Issue by tangjialin666** [ksyang2013/aimsgb#11]
I want to generate graphene grain boundary with [1 1 1] rotation axis, but it has been used for more than a day without any reaction. Are there any bugs？

**Comment by jic198** [comment]
@tangjialin666 I want to let you know that I am fully aware of this issue. For hexagonal input structure like graphene, the code only works for sigma 5, 13 and 17 GBs with [001] rotation axis. The most time consuming part is to find a transformation matrix to make each grain structure orthogonal, which is not guaranteed. That's why the code hangs. For those GBs, I am considering using `grain.set_orthogonal_grain` method. The only problem is that this method breaks the symmetry of the grain structure. I will have to let the users decide whether they want to use the generated GBs. I will add this option later and will keep you posted.

**Comment by tangjialin666** [comment]
Thanks for your reply. Hope I can get the new version of aimsgb soon.


JIalin Tang
2023.9.14



发自我的iPhone


------------------ Original ------------------
From: Jianli Cheng ***@***.***&gt;
Date: Thu,Sep 14,2023 1:53 PM
To: ksyang2013/aimsgb ***@***.***&gt;
Cc: tangjialin666 ***@***.***&gt;, Mention ***@***.***&gt;
Subject: Re: [ksyang2013/aimsgb] Can not generate graphene grain boundary with[1 1 1] rotation axis (Issue #11)





 
@tangjialin666 I want to let you know that I am fully aware of this issue. For hexagonal input structure like graphene, the code only works for sigma 5, 13 and 17 GBs with [001] rotation axis. The most time consuming part is to find a transformation matrix to make each grain structure orthogonal, which is not guaranteed. That's why the code hangs. For those GBs, I am considering using grain.set_orthogonal_grain method. The only problem is that this method breaks the symmetry of the grain structure. I will have to let the users decide whether they want to use the generated GBs. I will add this option later and will keep you posted.

**Comment by jic198** [comment]
@tangjialin666 I just pushed the latest update. Please note that for graphene grain boundary with [111] rotation axis, the new method breaks the symmetry in each grain structure. Please check the generated structures and let me know if you have additional questions.

**Comment by tangjialin666** [comment]
The z-axis of graphene or hexagonal boron nitride (POSCAR from Material Project) is perpendicular to the (001) plane, which makes it impossible to obtain reasonable tilted grain boundaries for (111) rotation axis. May I ask if the reasonable approach is to change the z-axis to be perpendicular to the (111) rotation axis before using the code to generate GBs？
Looking forward to your answer, thank you very much

**Comment by jic198** [comment]
I don't think that will work. My understanding is that graphene is a 2D structure. So, changing from [001] to [111] doesn't make much sense to me. Have you seen any references talking about [111] graphene structures?

**Comment by tangjialin666** [comment]
> I don't think that will work. My understanding is that graphene is a 2D structure. So, changing from [001] to [111] doesn't make much sense to me. Have you seen any references talking about [111] graphene grain boundary structures?

No, I have not seen any [111] graphene grain boundary structures before. The reason why I raised this question is that, for aimsgb, the rotation angle for hcp structure may be incorrect. For example, 9.43° is a typical rotation angle for graphene or any other hcp grain boundary, corresponding to sum 37, (7-3-4), [001]. However, for aimsgb (and the table A2 in cms_2018 paper), 9.43° corresponds to sum 37, (7-3-4), [111]. I suspect that aimsgb encountered an error when converting rotation axis form [UVW] to [uvtw]. Specifically, the rotation axis for hcp [0001] is equal to [001], but not [111].

**Comment by tangjialin666** [comment]
> I don't think that will work. My understanding is that graphene is a 2D structure. So, changing from [001] to [111] doesn't make much sense to me. Have you seen any references talking about [111] graphene grain boundary structures?

No, I have not seen any [111] graphene grain boundary structures before. The reason why I raised this question is that, for aimsgb, the rotation angle for hcp structure may be incorrect. For example, 9.43° is a typical rotation angle for graphene or hcp grain boundary, corresponding to sum 37, (7-3-4), [001]. However, for aimsgb (and the table A2 in cms_2018 paper), 9.43° corresponds to sum 37, (7-3-4), [111]. I suspect that aimsgb encountered an error when converting rotation axis form [UVW] to [uvtw]. Specifically, the rotation axis for hcp [0001] is equal to [001], but not [111].
